src/elm/Page/Device.elm

```diff
@@ -140,13 +142,32 @@ update session msg model =
 
         DeviceInfosDone (Err error) ->
             let
-                -- TODO handle error
                 ( message, details ) =
                     AstarteApi.errorToHumanReadable error
+
+                errorMessage =
+                    case error of
+                        AstarteApi.InvalidRequest ->
+                            "Invalid device ID."
```

Comment:
You might get 400 for several release, so this message is misleading. I would write "Invalid request."